Circle API key generation + management UI

META-DREAMER opened the pull request:
Currently available via URL only at `/admin/api`. Only allows setting up with read permissions for now.

## Test and Deployment Plan

Generate API keys with specific permissions and ensure that key can only fetch the data described in each permission and only for the circle it was generated for.

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/7143583/173030826-29083bb5-999b-4046-8f4e-7ed78b12bbdb.mp4


## Reviewers

<!-- Tag any reviewers who have context on this PR, or are familiar with this part of the codebase. -->

## Related Issue

#887

CryptoGraffe commented on src/pages/AdminCircleApiPage/constants.ts:
```diff
@@ -0,0 +1,15 @@
+import { circleApiPermissionsSelector } from './useCircleApiKeys';
+
+export const API_PERMISSION_LABELS: Record<
```
Can you help me understand what's going on here? What's the benefit of this type? There are other string literals (eg the explainer text) that isn't folded into a type, so I'm trying to understand what you were going for here? It seems nice but I feel like i'm missing something.

CryptoGraffe replied:
I guess this is mapping the keys of the selector to friendly strings?

META-DREAMER replied:
The purpose of typing it with a `Record` is to ensure that every possible value from circleApiPermissions has a corresponding friendly string. If I were to remove one of those lines, it would give a type error like this: 
![image](https://user-images.githubusercontent.com/7143583/176033440-1d8c5a98-cf6c-482a-92af-9a792426554e.png)


This makes sure that if we add any new permissions, we also add in a friendly string for it.

CryptoGraffe commented on src/pages/AdminCircleApiPage/ApiKeyForm.tsx:
```diff
@@ -0,0 +1,189 @@
+          infoTooltip={<>Allows reading information about circle nominees.</>}
+        />
+      </Box>
+      <Button
```
I think this button would look better taller, like on the Allocation flow , which is ```<UIButton
            size="large"
```

CryptoGraffe reviewed (approved):
Nice work this is great. Some of my comments are just to understand, cuz you are doing some neat patterns I'd like to learn from.

crabsinger commented:
I've merged main into this branch and resolved the merge conflicts. @META-DREAMER have you addressed @CryptoGraffe review?

META-DREAMER commented on api-lib/gql/__generated__/zeus/index.ts:
```diff
@@ -18268,17 +18269,17 @@ export type GraphQLTypes = {
-    update_allocations?: boolean | undefined;
+    read_pending_token_gifts?: boolean | undefined;
     update_circle?: boolean | undefined;
+    update_pending_token_gifts?: boolean | undefined;
```
probably because I renamed `update_allocations` to `update_pending_token_gifts` to make it more clear which table in the DB would be affected

META-DREAMER commented on src/lib/gql/__generated__/zeus/const.ts:
```diff
@@ -2937,6 +2941,11 @@ export const ReturnTypes: Record<string, any> = {
   },
+  GenerateApiKeyResponse: {
+    api_key: 'String',
+    circleApiKey: 'circle_api_keys',
```
Thought we were gonna camelCase any graphql stuff that we added to make it easier to differentiate between auto-generated hasura stuff. Can change to snake case if that's the pattern

META-DREAMER commented on src/pages/AdminCircleApiPage/AdminCircleApiPage.tsx:
```diff
@@ -0,0 +1,114 @@
+
+  const queryClient = useQueryClient();
+
+  const circleId = useSelectedCircleId();
```
Tried it, seems to be working fine. A bunch of places in the code are still using `useSelectedCircle`, which uses `useSelectedCircleId`. I just used it directly here since I dont need any of the other data that `useSelectedCircle` was assembling

META-DREAMER commented:
> I've merged main into this branch and resolved the merge conflicts. @META-DREAMER have you addressed @CryptoGraffe review?

Apologies for being late on this, just got back from my trip to NYC. Addressed feedback, ready for re-review now

crabsinger reviewed (approved):
lgtm - played with the staging build to create an api key as well
